packages/typespec-client-generator-core/src/package.ts

```diff
@@ -168,18 +173,55 @@ function getSdkPagingServiceMethod<TServiceOperation extends SdkServiceOperation
       });
     }
 
+    // eslint-disable-next-line @typescript-eslint/no-deprecated
     basic.response.resultPath = getPropertyPathFromModel(
       context,
       responseType?.__raw,
       (p) => p === pagingOperation.output.pageItems.property,
     );
-    const nextLinkPath = pagingOperation.output.nextLink
-      ? getPropertyPathFromModel(
-          context,
-          responseType?.__raw,
-          (p) => p === pagingOperation.output.nextLink!.property,
-        )
-      : undefined;
+    basic.response.resultSegments = getPropertySegmentsFromModelOrParameters(
```

Comment:
nit, var naming.

I need to go back 50 lines to see what `basic` is. And I still not have no idea why it is a "basic" from "getSdkBasicServiceMethod" -- is there an "advanced" from "getSdkAdvancedServiceMethod"? :-)

Reply:
paged operation just add more info into the basic operation.